engine-rest/engine-rest-openapi/README.md

```diff
@@ -0,0 +1,146 @@
+
+# Table of Contents
+1. [Build and phases](#build-and-phases)
+2. [Contribution](#contribution)
+TODO do we need it? if yes --> fill it up at the end
+
+## Build and phases
+
+To build the artifact run: `mvn clean install`
+
+The build contains:
+* Generation of formatted openapi.json file by parsing the [templates](./src/main/templates)
+* Validation of the generated file against [schema.json](./src/main/openapi/schema.json)
+* Generation of java client from the openapi.json file
+* Run of smoke tests against that client (the tests use [WireMock](http://wiremock.org/docs/))
+
+## Contribution
+
+NOTE: Please follow the next step to get familiar with the structure and the instructions that follow.
+
+For the generation of the documentation, we use [Freemarker](https://freemarker.apache.org/docs/index.html) templating language. The structure of the template is:
+```
+ +--main.ftl
+ +--lib
+ | +--macros.ftl
+ +--models
+ | +--org/camunda/bpm/engine/rest/dto
+ | | +--ExceptionDto.ftl
+ | | +--history
+ | | | +--HistoricProcessInstanceQueryDto.ftl
+ | | +--repository
+ | | | +--DeploymentDto.ftl
+ | | | +--ProcessDefinitionDto.ftl
+ | | +--runtime
+ | | | +--ActivityInstanceDto.ftl
+ +--paths
+ | +--commons
+ | | +--pagination-params.ftl
+ | | +--process-instance-query-params.ftl
+ | | +--sort-params.ftl
+ | | +--sort-props.ftl
+ | +--deployment
+ | | +--all.ftl
+ | | +--create
+ | | | +--post.ftl
+ | +--process-instance
+ | | +--all.ftl
+ | | +--get.ftl
+ | | +--suspended
+ | | | +--put.ftl
+ | | +--{id}
+ | | | +--delete.ftl
+ | | | +--variables
+ | | | | +--{varName}
+ | | | | | +--data
+ | | | | | | +--get.ftl
+ | | | | | | +--post.ftl
+ | | | | | +--delete.ftl
+```
+
+### main.ftl
+
+The `main.ftl` contains the general information of the OpenAPI doc:
+* OpenAPI spec version
+* info
+* externalDocs
+* servers (default and named)
+* tags (each resource has a tag)
+* wrap up of pats and components
+
+By parsing this file the end .json file is generated.
+
+### macros
+
+Contains all of the [macros](https://freemarker.apache.org/docs/ref_directive_macro.html) used in paths and models:
+* `parameter` (used in paths)
+* `property` (used in paths and models)
+* `requestBody` (used in paths)
+* `response` (used in paths)
+
+Try to get familiar with them as they are heavily used in the templates. Keep in mind that in certain situation, it is fine to not use them, if the endpoint/dto is too complex for example. Feel free to add missing parameters to the macros, however, do not forget to reflect your changes to all usages of the macros.
+Some parameters are required (`name` and `description` in most of the cases) and some need to be provided if necessary (nice to have - `minimum`,`defaultValue`, `deprecated`). 
+
+### models
+
+This folder contains all of the dto used in the request and response bodies. Instructions:
```

Comment:
I think, since it's an acronym, it should be with capital letters:
```suggestion
This folder contains all of the DTOs used in the request and response bodies. Instructions:
```